src/detect-engine-threshold.c

```diff
@@ -952,6 +952,8 @@ int PacketAlertThreshold(const DetectEngineCtx *de_ctx, DetectEngineThreadCtx *d
 {
     SCEnter();
 
+    uint8_t action = pa->action;
```

Comment:
const + comment like "keep original action to see if rate_filter changed it" would have helped me understand the logic